pennylane/ops/functions/equal.py

```diff
@@ -208,6 +210,21 @@ def _equal_operators(
     return getattr(op1, "inverse", False) == getattr(op2, "inverse", False)
 
 
+@_equal.register
+# pylint: disable=unused-argument, protected-access
+def _equal_prod_and_sum(op1: CompositeOp, op2: CompositeOp, **kwargs):
+    """Determine whether two Prod or Sum objects are equal"""
+
+    # organizes by wire indicies while respecting commutation relations
+    sorted_ops1 = op1._sort(op1.operands)
+    sorted_ops2 = op2._sort(op2.operands)
+
+    if len(sorted_ops1) != len(sorted_ops2):
+        return False
+
+    return np.all([qml.equal(o1, o2, **kwargs) for o1, o2 in zip(sorted_ops1, sorted_ops2)])
```

Comment:
```suggestion
    return all(equal(o1, o2, **kwargs) for o1, o2 in zip(sorted_ops1, sorted_ops2))
```
This should work at least.

This way, Python will stop iterating when it encounters the first False. We dont want to do more work then we need to.

Reply:
Thanks!